### PR TITLE
first example for idea extended devices.

### PR DESCRIPTION
**Pull request by Fjuxx**
This was my idea in general (not tested any code, so i probably broke the plugin with this commit ;))

But here you can see what direction i want to go.

This also seems to be the way pimatic (internally) is set up.

For example. The dimmer class is extending the class of the switch. So you do not have to implement On/Off or Toggle functions. Only 1 change state function and all of the above work as well.

Let me know what you think!

### CODE REVIEW

File: pimatic-led-light.coffee [hunk 1]
```diff
@@ -17,9 +17,9 @@ module.exports = (env) ->
     init: (app, @framework, @config) =>
       deviceConfigDef = require("./led-light-schema")
 
-      @framework.deviceManager.registerDeviceClass "LedLight",
+      @framework.deviceManager.registerDeviceClass "IWYLed",
```
Comment (philip1986) [inline pimatic-led-light.coffee:20]: MilightLed should be also attached to the framework here!?

Comment (Fjuxx) [inline pimatic-led-light.coffee:20]: Yes. But please Note. I just made this for The general idea. So i probably broke almost erything with this pull ;)

Comment (philip1986) [inline pimatic-led-light.coffee:20]: Yes, I aware that this is a blueprint. I just wanted to prof that I got it right.

Comment (Fjuxx) [inline pimatic-led-light.coffee:20]: Yea you got it right ;)

So only The plugin itself is seperate. All of The action providers ect. And The gui is shared

Comment (philip1986) [inline pimatic-led-light.coffee:20]: Ok, I like this idea.

**Comment by philip1986** [comment]
I've opened a new PR based on this https://github.com/philip1986/pimatic-led-light/pull/7

**Pull request closed without merging**